pokemongo_bot/cell_workers/utils.py

```diff
@@ -144,3 +145,57 @@ def print_yellow(message):
 
 def print_red(message):
     print(u'\033[91m' + message.decode('utf-8') + '\033[0m')
+
+def nickname_pokemon(bot,pokemon):
+    """This requies a pokemon object containing all the standard fields: id, ivs, cp, etc"""
+    template = bot.config.nickname_template.lower()
+    if template == "{name}":
```

Comment:
in a manner of speaking, yes. the `pokemon` object may or may not have a `'nickname'` key in it, depending on if it's been nicknamed previously or not.